pkg/proc/core/linux_core.go

```diff
@@ -62,45 +113,39 @@ func readLinuxAMD64Core(corePath, exePath string) (*Process, error) {
 		return nil, fmt.Errorf("%v is not an exe file", exeELF)
 	}
 
-	notes, err := readNotes(coreFile)
+	machineType := exeELF.Machine
+	notes, err := readNotes(coreFile, machineType)
 	if err != nil {
 		return nil, err
 	}
 	memory := buildMemory(coreFile, exeELF, exe, notes)
 	entryPoint := findEntryPoint(notes)
-
 	p := &Process{
 		mem:         memory,
 		Threads:     map[int]*Thread{},
 		entryPoint:  entryPoint,
-		bi:          proc.NewBinaryInfo("linux", "amd64"),
 		breakpoints: proc.NewBreakpointMap(),
 	}
 
-	var lastThread *linuxAMD64Thread
-	for _, note := range notes {
-		switch note.Type {
-		case elf.NT_PRSTATUS:
-			t := note.Desc.(*LinuxPrStatus)
-			lastThread = &linuxAMD64Thread{linutil.AMD64Registers{Regs: &t.Reg}, t}
-			p.Threads[int(t.Pid)] = &Thread{lastThread, p, proc.CommonThread{}}
-			if p.currentThread == nil {
-				p.currentThread = p.Threads[int(t.Pid)]
-			}
-		case NT_X86_XSTATE:
-			if lastThread != nil {
-				lastThread.regs.Fpregs = note.Desc.(*linutil.AMD64Xstate).Decode()
-			}
-		case elf.NT_PRPSINFO:
-			p.pid = int(note.Desc.(*LinuxPrPsInfo).Pid)
-		}
+	switch machineType {
```

Comment:
Since we're checking machine type here, we should return an error if we get an unsupported machine type.